samples/client-provider/query.fsx

```diff
@@ -0,0 +1,37 @@
+﻿#r "../../src/FSharp.Data.GraphQL.Client/bin/Debug/FSharp.Data.GraphQL.Client.dll"
+
+open FSharp.Data.GraphQL
+open System.Collections.Generic
+
+let [<Literal>] serverUrl = "http://localhost:8083"
+
+// The name and arguments of the query will be automatically set by the type provider
+let [<Literal>] queryFields = "{ id, name, appearsIn, friends { name } }"
+let [<Literal>] queryFieldsWithFragments = "{ ...data, friends { name } } fragment data on Human { id, name, appearsIn }"
+
+type MyClient = GraphQLProvider<serverUrl>
+
+let hero =
+    MyClient.QueryHero<queryFieldsWithFragments>("1000")
+    |> Async.RunSynchronously
+
+let droid =
+    MyClient.QueryDroid<queryFields>("2000")
+    |> Async.RunSynchronously
+
+// Result is an option type
+match hero with
+| None -> ()
+| Some hero ->
+    printfn "My hero is %A" hero.name
+    printfn "Appears in %O: %b" MyClient.Episode.Empire
+        (hero.appearsIn |> Array.exists ((=) MyClient.Episode.Empire))
```

Comment:
This is easy to do and will be safer indeed. BTW, I've a question about Enums: currently I'm using the value names as the possible values as I cannot find a `Value` field in `IntrospectionEnumVal`. According to [this](http://graphql.org/docs/api-reference-type-system/#graphqlenumtype), _GraphQL serializes Enum values as strings, however internally Enums can be represented by any kind of type_. Is this _internal representation_ only happening in the server and enum values are always sent to the client represented with their string names.. In other words, it's [my current approach](https://github.com/alfonsogarciacaro/FSharp.Data.GraphQL/blob/client-provider/src/FSharp.Data.GraphQL.Client/IntrospectionParser.fs#L145-L148) right? Note, for example, [I had to modify the case of enum values](https://github.com/alfonsogarciacaro/FSharp.Data.GraphQL/commit/4f0d5a017b610768a2635c164d8143c0d1685e6c#diff-e55cde849b1089690809ecfacdc8847fL92) in the `server.fsx` example to make it work right.